signup/tool/src/java/org/sakaiproject/signup/tool/jsf/UserLocale.java

```diff
@@ -34,39 +39,49 @@
  * @author Peter Liu
  */
 
+@Slf4j
 public class UserLocale {
 
 	private ResourceLoader rb = new ResourceLoader("messages");
-	private SakaiFacade sakaiFacade;
+
+	@Getter @Setter private SakaiFacade sakaiFacade;
 
 	public String getLocale() {
 		return (String) this.rb.getLocale().toString();
 	}
-
-	public SakaiFacade getSakaiFacade() {
-		return sakaiFacade;
-	}
-
-	public void setSakaiFacade(SakaiFacade sakaiFacade) {
-		this.sakaiFacade = sakaiFacade;
-	}
 	
 	/**
 	 * Get the date format from the locale
 	 * @return
 	 */
 	public String getDateFormat() {
-		DateFormat df = DateFormat.getDateInstance(DateFormat.SHORT, (new ResourceLoader()).getLocale());
-		return ((SimpleDateFormat)df).toPattern();
+
+		Locale locale = new ResourceLoader().getLocale();
+		DateFormat df = DateFormat.getDateInstance(DateFormat.SHORT, locale);
+		try {
+			return ((SimpleDateFormat)df).toPattern();
+		} catch (ClassCastException cce) {
+			log.warn("Failed to cast DateFormat into SimpleDateFormat for locale {}", locale.toString());
+			return new SimpleDateFormat().toPattern();
+		}
 	}
 
 	/**
-	 * Get the date format from the locale with short timezone at end
+	 * Get the date format from the locale
 	 * @return String representing a SimpleDateFormat for use with JSF convertDateTime
 	 */
 	public String getFullDateTimeFormat() {
-		DateFormat df = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG, this.rb.getLocale());
-		return ((SimpleDateFormat)df).toPattern();
+
+		Locale locale = this.rb.getLocale();
+		DateFormat df = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.SHORT, locale);
+		try {
+			SimpleDateFormat sdf = (SimpleDateFormat) df;
+			sdf.applyLocalizedPattern("MMMM d, yyyy h:mm a");
```

Comment:
a custom format doesn't seem right to me...... why doesn't the DateFormat above work?

Reply:
It ships the default full format, with timezone

Reply:
If you look at the docs for DateFormat, it explains that if you want to change the default, you have to cast to a SimpleDateFormat and customise.

Reply:
You can tune DateFormat to a degree, by using SHORT, MEDIUM, LONG etc, but that's all.

Reply:
Still just seems more preferable to use a pre-defined format. For example, in your format, "a" means am/pm, right? 

So let's just use MEDIUM instead of LONG to avoid timezone

Reply:
Then you lose am pm

Reply:
I thought you'd be able to merge formats, like, just merge am pm into MEDIUM

Reply:
Sweden doesn't want am/pm. They want 24 hr time

Reply:
SHORT was actually the one to go for, for the time component. I've tested in Swedish, and PST, and this PR fixes that bunch of Signup timezone tickets.